lib/dynamoid/transaction/create.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'action'
+
+module Dynamoid
+  module Transaction
+    class Create < Action
+
+      def run_callbacks
+        (yield; return) unless model
+        model.run_callbacks(:save) do
+          model.run_callbacks(:create) do
+            model.run_callbacks(:validate) do
+              yield
+            end
+          end
+        end
+      end
+
+      def to_h
+        item = attributes.clone
+        item = add_timestamps(item)
+        item = model.attributes.merge(item) if model
+        item[model_class.hash_key] ||= SecureRandom.uuid
+        item = Dynamoid::Dumping.dump_attributes(item, model_class.attributes)
+
+        condition = "attribute_not_exists(#{model_class.hash_key})"
+        condition += " and attribute_not_exists(#{model_class.range_key})" if model_class.range_key? # needed?
+        {
+          put: {
+            item: sanitize_item(item),
+            table_name: model_class.table_name,
+            condition_expression: condition,
+          }
+        }
+      end
```

Comment:
After persisting a new model should become `persisted?` and modified fields should become not-changed:

https://github.com/Dynamoid/dynamoid/blob/6d53f100f0478bc04c7c45646771182df603fdc8/lib/dynamoid/persistence/save.rb#L46
https://github.com/Dynamoid/dynamoid/blob/6d53f100f0478bc04c7c45646771182df603fdc8/lib/dynamoid/dirty.rb#L47

Reply:
changes_applied is now called on the model immediately after processing the model. This is earlier than the actual commit. We could wait and call changes_applied on all the models after the commit.

Reply:
Yeah, I believe `changes_applied` should be called only after successful `TransactWriteItems` call.

Reply:
Regarding `persisted?` - I haven't noticed where a new model is being marked as persisted.

Reply:
new_record is now set to false after the commit.

I added an accessor for destroyed so that can be set when destroying.